fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/bitmap/ComposedBitmapIndexAggregate.java

```diff
@@ -73,6 +73,39 @@
         this.root = root;
     }
 
+    /**
+     * Try to build a composed bitmap plan for the given query and aggregate function.
+     * @param planner a query planner to use to construct the plans
+     * @param query a query providing target record type, filter, and required fields
+     * @param indexAggregateFunction the function giving the desired position and grouping
+     * @return an {@code Optional} query plan or {@code Optional.empty} if planning is not possible
+     */
+    @Nonnull
+    public static Optional<ComposedBitmapIndexQueryPlan> tryPlan(@Nonnull RecordQueryPlanner planner,
+                                                                 @Nonnull RecordQuery query,
+                                                                 @Nonnull IndexAggregateFunction indexAggregateFunction) {
+        if (query.getFilter() == null) {
+            return Optional.empty();
+        }
```

Comment:
Should this also reject the query if `query.getSort() != null`?

Reply:
Yes, that seems like good added safety.